Sources/NetworkProtection/ExtensionMessage/ExtensionMessage.swift

```diff
@@ -18,17 +18,199 @@
 
 import Foundation
 
-public enum ExtensionMessage: UInt8 {
-    case resetAllState = 0
+public enum ExtensionMessage: RawRepresentable {
```

Comment:
refactored the Message encoding/decoding in sake of encapsulation and adding new messages